src/utils/crypto_utils/src/crypto_lib.c

```diff
@@ -441,3 +458,139 @@ void CryptoUtils_FreeCryptoKeyHandle(CryptoKeyHandle key)
 {
     EVP_PKEY_free(CryptoKeyHandleToEVP_PKEY(key));
 }
+
+/**
+ * @brief Computes a SHA256 hash from bytes.
+ *
+ * @param buf The handle to the input bytes to be hashed.
+ * @return CONSTBUFFER_HANDLE The digest buffer handle. Caller will own it and need to call CONSTBUFFER_DecRef() to free.
+ */
+CONSTBUFFER_HANDLE CryptoUtils_CreateSha256Hash(const CONSTBUFFER_HANDLE buf)
+{
+    const size_t Sha256DigestNumBytes = 32;
+    unsigned char hashBuf[Sha256DigestNumBytes];
```

Comment:
is this assumed not be to null terminated? does 32 accountfor null termination?

Reply:
Correct. It is not null-terminated string but buffer of exactly 32 _unsigned_ char bytes, so exactly 32 bytes.